fendermint/app/src/cmd/run.rs

```diff
@@ -59,9 +79,21 @@ async fn run(settings: Settings) -> anyhow::Result<()> {
         state_store,
         interpreter,
         resolve_pool,
-        parent_finality,
+        parent_finality_provider.clone(),
     )?;
 
+    if settings.ipc.is_topdown_enabled() {
+        let topdown_config = settings.ipc.top_down_config()?;
```

Comment:
```suggestion
        let topdown_config = settings.ipc.top_down_config().context("failed to parse IPC config")?;
```

Although it would also be nice to find a way to combine this with the previous call to be honest, where the same `is_topdown_enabled()` was followed by this. You could turn it into an optional configuration, and then use that to drive the decision:

```
let topdown_config = if settings.ipc.is_topdown_enabled() { 
  Some(settings.ipc.top_down_config().context("failed to parse topdown config")?)
else { 
  None
};

if let Some(topdown_config) = topdown_config { <create-provider> }
...
if let Some(topdown_config) = topdown_config { <start-syncer> }
```

Reply:
yep, updated to something along this line.